great_expectations/datasource/fluent/data_asset/path/directory_asset.py

```diff
@@ -66,10 +107,24 @@ def _get_batch_definition_list(
             )
         return batch_definition_list
 
-    @public_api
-    def add_batch_definition_whole_directory(self, name: str) -> BatchDefinition:
-        """Add a BatchDefinition which creates a single batch for the entire directory."""
-        return self.add_batch_definition(name=name, partitioner=None)
+    @singledispatchmethod
```

Comment:
i like the syntax, but tend to avoid it because typing always gets wonky. not sure what is going on in the implementation but mypy has a very difficult time reasoning about what type is being returned. Also, it's not possible to use keyword args in the signature, which is unfortunate.